website/docs: Update docs for single logout (#17169)
goauthentik/authentik: website/docs/add-secure-apps/providers/single-logout/index.md

---
title: Single Logout (SLO)
---

> Single Logout is a security feature that logs a user out of all their applications with active sessions when they log out of authentik. It uses the OAuth2/OpenID Connect front-channel and back-channel logout specifications in combination with SAML's Single Logout specification.

Reviewer: ```suggestion
Single Logout (SLO) is a security feature that logs users out of all active applications when they log out of authentik. It uses the OAuth2/OpenID Connect front-channel and back-channel logout specifications in combination with SAML's Single Logout specification.
```

For example, if a user is concurrently logged into an OIDC application and two SAML applications, when the user logs out of authentik, they will automatically be logged out of all three applications. Without SLO configured, users with active sessions across multiple applications would need to manually log out of each one.

:::info
Check with your service provider to see if they support SAML Single Logout or OIDC front-channel/back-channel logout. Not all service providers support these features.
:::

## How Single Logout works in authentik

When a user logs out or their session is terminated in authentik, the following process occurs:

1. **Session Termination**: The user's session ends through one of the following:
    - User-initiated logout via a logout flow
    - Administrative action (session deletion or user deactivation)
    - Token revocation or expiration

> 2. **Provider Identification**: authentik identifies all OAuth2/OIDC and SAML providers with active sessions for the user that have SLO configured

Reviewer: ```suggestion
2. **Provider Identification**: authentik identifies all OAuth2/OIDC and SAML providers with active sessions that have SLO configured for the user.
```

Front-channel logout sends logout requests through the user's browser. authentik supports two front-channel modes:

#### iframe mode (default for OIDC)

    - Loads all provider logout URLs simultaneously in hidden iframes
    - Provides fast, parallel logout across multiple providers
    - Required by the OIDC front-channel logout specification
    - Most SAML providers also support iframe-based logout
    - Provides fast, parallel logout across multiple providers
    - Required by the OIDC front-channel logout specification
    - Most SAML providers also support iframe-based logout

#### Native Mode (SAML Only)

    - Uses the active browser tab to chain redirects and POST requests sequentially
    - Provides better compatibility with SAML providers that have iframe restrictions
    - Each provider redirects the user back to authentik before proceeding to the next provider
    - Not available for OIDC providers as the specification requires iframe support

:::info
Use native front-channel mode for SAML providers if you encounter iframe compatibility issues, such as Content Security Policy (CSP) restrictions or cookie handling problems.
:::

### Back-channel Logout

> Back-channel logout sends logout requests directly from the authentik server to the provider's logout endpoint via HTTP POST.

Reviewer: ```suggestion
Back-channel logout sends logout requests directly from the authentik server to each provider's logout endpoint via HTTP POST.
```
